packages/flutter_tools/lib/src/commands/symbolize.dart

```diff
@@ -108,47 +151,71 @@ class SymbolizeCommand extends FlutterCommand {
       input = _stdio.stdin;
     }
 
-    String debugInfoPath = stringArg('debug-info')!;
-
-    // If it's a dSYM container, expand the path to the actual DWARF.
-    if (debugInfoPath.endsWith('.dSYM')) {
-      final Directory debugInfoDir = _fileSystem
-        .directory(debugInfoPath)
-        .childDirectory('Contents')
-        .childDirectory('Resources')
-        .childDirectory('DWARF');
-
-      final List<FileSystemEntity> dwarfFiles = debugInfoDir.listSync().whereType<File>().toList();
-      if (dwarfFiles.length == 1) {
-        debugInfoPath = dwarfFiles.first.path;
-      } else {
-        throwToolExit('Expected a single DWARF file in a dSYM container.');
+    final Map<int, String> debugInfoPathMap = _unitDebugInfoPathMap();
+    final Map<int, Uint8List> unitSymbols = <int, Uint8List>{};
+
+    for (final int unitId in debugInfoPathMap.keys) {
+      String debugInfoPath = debugInfoPathMap[unitId]!;
+
+      // If it's a dSYM container, expand the path to the actual DWARF.
+      if (debugInfoPath.endsWith('.dSYM')) {
+        final Directory debugInfoDir = _fileSystem
```

Comment:
Call me paranoid, but can we check that `debugInfoDir` exists, and then throw some kind of error if it does not? Maybe a `StateError`? I've learned not to assume anything about any file system operation on users' machines.

Reply:
There's a `MachO.handleDSYM' in `pkg:native_stack_traces` (also not exported) that could be used here instead of manually recreating the same checks. I'll also export that.

Reply:
Actually, it uses `dart:io` directly instead of going through the `FileSystem` abstraction, so changing my mind on doing this.

Reply:
However, that leaves the question as to how to appropriately check this using the `package:file` interface, but I imagine that `childDirectory` and `listSync` do enough checking and throwing if invalid that we're covered for this case? (We already check the path is a directory before turning it into a `Directory` and then calling `childDirectory` on it.)

Reply:
SGTM